docs/core/porting/cppcli.md

```diff
@@ -0,0 +1,88 @@
+4. Update API usage in cpp files, as necessary, to remove APIs unavailable to .NET Core. Because C++/CLI projects tend to be fairly thin interop layers, there are often not many API changes needed.
+    1. The [.NET Portability Analyzer](https://docs.microsoft.com/dotnet/standard/analyzers/portability-analyzer) can be used to identify unsupported .NET APIs used by C++/CLI binaries just as with purely managed binaries. Guidelines for determining code portability and updating projects to work with .NET Core APIs are available in the [library porting guidance](./libraries.md#determining-the-portability-of-your-code).
+
+### WPF and Windows Forms usage
+
+C++/CLI projects that use Windows Forms or WPF APIs can be ported to .NET Core, but need explicit framework references to the UI libraries. SDK-style projects that use Windows desktop APIs reference the necessary framework libraries automatically by using the `Microsoft.NET.Sdk.WindowsDesktop` SDK. Because C++/CLI projects don't use the SDK-style project format, they need to add explicit framework references when targeting .NET Core.
+
+To use Windows Forms APIs, add this reference to the vcxproj:
+
+```xml
+<!-- Reference all of Windows Forms -->
+<FrameworkReference Include="Microsoft.WindowsDesktop.App.WindowsForms" />
+```
+
+To use WPF APIs, add this reference to the vcxproj:
+
+```xml
+<!-- Reference all of WPF -->
+<FrameworkReference Include="Microsoft.WindowsDesktop.App.WPF" />
+```
+
+To use both Windows Forms and WPF APIs, add this reference to the vcxproj:
+
+```xml
+<!-- Reference the entirety of the Windows desktop framework:
+     Windows Forms, WPF, and the types that provide integration between them -->
+<FrameworkReference Include="Microsoft.WindowsDesktop.App" />
+```
+
+Currently, it is not possible to add these references using Visual Studio's reference manager. The project file needs to be updated manually.
+
+## Building without MSBuild
+
+The MSBuild project system makes it easy to port a vcxproj-based project to .NET Core. If you need to build a C++/CLI project for .NET Core directly with cl.exe and link.exe, it can be done with a few additional steps.
+
+1. When compiling, pass `/clr:netcore` to cl.exe.
+2. Reference necessary .NET Core reference assemblies.
+3. When linking, provide the .NET Core app host directory as a LibPath (so that ijwhost.lib can be found).
+4. Copy ijwhost.dll (from the .NET Core app host directory) to the project's output directory.
+5. Make sure a [.runtimeconfig.json](https://github.com/dotnet/cli/blob/master/Documentation/specs/runtime-configuration-file.md) file exists for the first component of the application that will run managed code. If the application has a managed entry point, this will be created and copied automatically. If the application has a native entry point, though, and a C++/CLI library will be the first time the process uses the .NET Core runtime, you will need to create a runtimeconfig.json file for the library.
```

Comment:
```suggestion
5. Make sure a [.runtimeconfig.json](https://github.com/dotnet/cli/blob/master/Documentation/specs/runtime-configuration-file.md) file exists for the first component of the application that will run managed code. If the application has a managed entry point, this will be created and copied automatically. If the application has a native entry point, though, and a C++/CLI library will be the first time the process uses the .NET Core runtime, you will need to create a `runtimeconfig.json` file for the library.
```